### ISSUE TITLE
lexer incorrectly reports span starts for JSX children

### ISSUE THREAD

**Issue by camc314** [web-infra-dev/oxc#1026]
See on https://web-infra-dev.github.io/oxc/playground/?code=3YCAAICngICAgICAgICxG0qZRraXI4KRyqLYsbIYJC83n4fDS3zVfaDS2lG%2F4ESQZs68lv%2FFKAIZf39GqsCA

Repro:

```jsx
const foo = () => {
  return <>{a}</>
}
```

The issue:

```json
{
// Ommited for brevity
                    "closingFragment": {
                      "type": "JSXClosingFragment",
                      "start": 0, // Incorrect
                      "end": 37
                    },
                    "children": [
                      {
                        "type": "JSXExpressionContainer",
                        "start": 0, // incorrect
                        "end": 34,
                        "expression": {
                          "type": "IdentifierReference",
                          "start": 32,
                          "end": 33,
                          "name": "a"
                        }
                      }
                    ]
}
```

**Comment by Boshen** [comment]
I managed to find the spot where we need to add the code, but I need to be afk now so I'll leave it here if you want to pick this up. Don't spend too much time on this if you can't figure this out, I'm also finding it hard to understand what's going on here.

```rust
    /// Re-tokenize for JSX child
    pub fn next_jsx_child(&mut self) -> Token<'a> {
        self.current.token.start = self.offset(); // plus or minus some offset similar to all the re-tokenize functions
        let kind = self.read_jsx_child();
        self.finish_next(kind)
    }
```
